presto-plugin-toolkit/src/main/java/io/prestosql/plugin/base/type/DateTimes.java

```diff
@@ -11,7 +11,7 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
-package io.prestosql.type;
+package io.prestosql.plugin.base.type;
```

Comment:
Let's not move this class to the plugin toolkit. In fact, presto-main should not depend on plugin-toolkit at all. Once all the variable-precision datetime PRs go in, we need to do a round of cleanup of all the shared code and figure out what things should go where. For now, please copy the necessary code into the plugin or plugin toolkit and add a TODO indicating it's a duplicate of code in this class.

Reply:
>  In fact, presto-main should not depend on plugin-toolkit at all. 

Sounds reasonable. Then we need some other place for shared code between engine and (our) connectors. 
Otherwise we will probably be putting more things in SPI than we otherwise would like to.

Reply:
I added duplicate functions to `io.prestosql.plugin.kafka.encoder.json.format.util.TimeConversions` with TODOs